src/kv/genericserialisewrapper.h

```diff
@@ -196,10 +196,26 @@ namespace kv
       // writer.
       if (!crypto_util)
       {
+        if (serialised_public_domain.size() <= 1)
```

Comment:
At the moment, the constructor of `GenericSerialiseWrapper` serialises the version but it would be easy to either 1) serialise the version after the `GenericSerialiseWrapper` is created (we should enforce that the version should be serialised before anything else) or 2) add a state enum to detect whether only the version has been serialised. I think 2 is neater and more extendable. 

Then, you can `return {}` based on that enum.

Reply:
Thanks I will think about this again. I looked into it a little bit. There is the minor caveat that we also have a `remove` function in `GenericSerialiseWrapper` so you would need to make checks on `remove` too in order to make sure that you still have data that you want to serialise, which makes it a bit less neater. You will still need to check if only the version is left in the end.

Reply:
These removes aren't removing things from the serialised data, they're serialising removals (a kind of write?) from the kv. I think the minimal enum solution here is to add a `bool ignore = false` arg to `serialise_internal`. When we call `serialise_internal(version_)` from the constructor, we can ignore this and don't change our `has_contents` enum, but any _other_ calls to `serialise_internal` will try to set the enum.

Reply:
Implemented with a `bool` arg on `serialise_internal` and a `bool` `has_contents` field